ros_buildfarm/common.py

```diff
@@ -644,3 +646,30 @@ def get_implicitly_ignored_package_names(cached_pkgs, explicitly_ignored_pkg_nam
         break
 
     return ignored_pkg_names.difference(explicitly_ignored_pkg_names)
+
+
+def get_package_conditional_context(index, rosdistro_name):
+    python_version = index.distributions[rosdistro_name].get('python_version')
+    ros_version = {
+        'ros1': '1',
+        'ros2': '2',
+    }.get(index.distributions[rosdistro_name].get('distribution_type'))
+
+    condition_context = {
+        'ROS_DISTRO': rosdistro_name,
+    }
+
+    if python_version:
+        condition_context['ROS_PYTHON_VERSION'] = python_version
+    if ros_version:
+        condition_context['ROS_VERSION'] = ros_version
+
+    return condition_context
+
+
+def evaluate_package_conditionals(index, rosdistro_name, packages):
+    condition_context = get_package_conditional_context(index, rosdistro_name)
+    for pkg in packages:
+        if pkg.package_format < 3:
+            continue
```

Comment:
I don't think you need this conditional check. The API allows you to call `evaluate_conditions` on any package.